### PR TITLE
Improve Backend Roadmap.

### PR DESCRIPTION
**Pull request by offensive-vk**
## Improves the Overall Backend Dev Roadmap.

- Fix Typos and Style
- Add Content
- Add Links
- Reorder Links
- Reorder Content
- Fix Formatting and EOL
- Change Anchors and Link Types.

This PR is intended to enhance the overall backend dev roadmap.

Current Status : **Finished** ⚒️

### CODE REVIEW

File: src/data/roadmaps/backend/content/bcrypt@dlG1bVkDmjI3PEGpkm1xH.md [hunk 1]
```diff
@@ -1,9 +1,9 @@
...
+Bcrypt is a password-hashing function designed to securely hash passwords for storage in databases. Created by **Niels Provos** and **David Mazières**, it's based on the Blowfish cipher and incorporates a salt to protect against rainbow table attacks. Bcrypt's key feature is its adaptive nature, allowing for the adjustment of its cost factor to make it slower as computational power increases, thus maintaining resistance against brute-force attacks over time. It produces a fixed-size hash output, typically 60 characters long, which includes the salt and cost factor. Bcrypt is widely used in many programming languages and frameworks due to its security strength and relative ease of implementation. Its deliberate slowness in processing makes it particularly effective for password storage, where speed is not a priority but security is paramount.
```
Comment (dansholds) [inline src/data/roadmaps/backend/content/bcrypt@dlG1bVkDmjI3PEGpkm1xH.md:3]:
```suggestion
Bcrypt is a password-hashing function designed to securely hash passwords for storage in databases. Created by Niels Provos and David Mazières, it's based on the Blowfish cipher and incorporates a salt to protect against rainbow table attacks. Bcrypt's key feature is its adaptive nature, allowing for the adjustment of its cost factor to make it slower as computational power increases, thus maintaining resistance against brute-force attacks over time. It produces a fixed-size hash output, typically 60 characters long, which includes the salt and cost factor. Bcrypt is widely used in many programming languages and frameworks due to its security strength and relative ease of implementation. Its deliberate slowness in processing makes it particularly effective for password storage, where speed is not a priority but security is paramount.
```

File: src/data/roadmaps/backend/content/caddy@Op-PSPNoyj6Ss9CS09AXh.md [hunk 2]
```diff
@@ -5,5 +5,5 @@ Caddy is a modern, open-source web server written in Go. It's known for its simp
 Visit the following resources to learn more:
 
 - [@official@Official Website](https://caddyserver.com/)
-- [@opensource@caddyserver/caddy](https://github.com/caddyserver/caddy)
+- [@opensource@Caddy on GitHub](https://github.com/caddyserver/caddy)
```
Comment (dansholds) [inline src/data/roadmaps/backend/content/caddy@Op-PSPNoyj6Ss9CS09AXh.md:8]:
```suggestion
- [@opensource@caddyserver/caddy - Caddy on GitHub](https://github.com/caddyserver/caddy)
```

File: src/data/roadmaps/backend/content/loadshifting@HoQdX7a4SnkFRU4RPQ-D5.md [hunk 3]
```diff
@@ -4,4 +4,5 @@ Load shifting is a strategy used to manage and distribute computing or system wo
 
 Learn more from the following resources:
 
+- [@article@Load Shifting - Wikipedia](https://en.wikipedia.org/wiki/Load_shifting)
```
Comment (dansholds) [inline src/data/roadmaps/backend/content/loadshifting@HoQdX7a4SnkFRU4RPQ-D5.md:7]:
```suggestion
```

Comment (dansholds) [reply src/data/roadmaps/backend/content/loadshifting@HoQdX7a4SnkFRU4RPQ-D5.md:7]: This is about load shifting on shifts, not networking systems

File: src/data/roadmaps/backend/content/ms-sql@dEsTje8kfHwWjCI3zcgLC.md [hunk 4]
```diff
@@ -5,6 +5,6 @@ Microsoft SQL Server (MS SQL) is a relational database management system develop
 Visit the following resources to learn more:
 
 - [@roadmap@SQL Roadmap](https://roadmap.sh/sql)
-- [@article@MS SQL website](https://www.microsoft.com/en-ca/sql-server/)
+- [@official@MS-SQL](https://www.microsoft.com/en-ca/sql-server/)
```
Comment (dansholds) [inline src/data/roadmaps/backend/content/ms-sql@dEsTje8kfHwWjCI3zcgLC.md:8]:
```suggestion
- [@official@MS SQL](https://www.microsoft.com/en-ca/sql-server/)
```

File: src/data/roadmaps/backend/content/open-api-specs@9cD5ag1L0GqHx4_zxc5JX.md [hunk 5]
```diff
@@ -7,4 +7,4 @@ Visit the following resources to learn more:
 - [@official@OpenAPI Specification Website](https://swagger.io/specification/)
 - [@official@Open API Live Editor](https://swagger.io/tools/swagger-editor/)
 - [@video@OpenAPI 3.0: How to Design and Document APIs with the Latest OpenAPI Specification 3.0](https://www.youtube.com/watch?v=6kwmW_p_Tig)
-- [@vidoe@ REST API and OpenAPI: It’s Not an Either/Or Question ](https://www.youtube.com/watch?v=pRS9LRBgjYg)
+- [@vidoe@ REST API and OpenAPI: It’s Not an Either/Or Question](https://www.youtube.com/watch?v=pRS9LRBgjYg)
```
Comment (dansholds) [inline src/data/roadmaps/backend/content/open-api-specs@9cD5ag1L0GqHx4_zxc5JX.md:10]:
```suggestion
- [@video@ REST API and OpenAPI: It’s Not an Either/Or Question](https://www.youtube.com/watch?v=pRS9LRBgjYg)
```

File: src/data/roadmaps/backend/content/twelve-factor-apps@8DmabQJXlrT__COZrDVTV.md
Comment (dansholds) [inline src/data/roadmaps/backend/content/twelve-factor-apps@8DmabQJXlrT__COZrDVTV.md:1]: You've removed 2 of the 12 methods?

**Review by dansholds — CHANGES_REQUESTED** [review]
Looks good, I've left a few comments to address.

File: src/data/roadmaps/backend/content/twelve-factor-apps@8DmabQJXlrT__COZrDVTV.md
Comment (offensive-vk) [reply src/data/roadmaps/backend/content/twelve-factor-apps@8DmabQJXlrT__COZrDVTV.md:1]: yeah, so many points 🤣

**Pull request merged**